### PR TITLE
[OTP] implemented sign up otp, some styling

### PR DESCRIPTION
**Pull request by philipye314**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Implementing the OTP flow for user SignUp
- Used another SMTP to get around rate limiting, now confirm user is turned on in Supabase.

### Screenshots

  

basic-otp-email

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- this is just a beginning test, so try making a new user, verifying it, and proceeding with sign-up as usual. Then, check their settings to make sure everything works as it used to work. I haven't connected this flow to Reset/Forgot password yet, so please don't try that yet.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- have a lot of questions regarding the flows --> specifically how Signing up a user for the first time will have some weird behavior due to creating a "half-baked" user. Guess we'll talk about it during team meeting
(leaving this note for myself: when I first signup a user using otp, pass in the name as well as a parameter. that way I don't have to store this in the otp flow, which would minimize more state, since it doesn't make sense for the Forgot/Reset password flows to require a name.)

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

[Supabase OTP Docs](https://supabase.com/docs/reference/javascript/auth-verifyotp)

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[GWN OTP Flow](https://github.com/calblueprint/girls-write-now/pull/25)

CC: @ronniebeggs

### CODE REVIEW

File: src/app/(Authentication)/SignUp/Email/index.tsx [hunk 1]
```diff
@@ -42,10 +43,17 @@ export default function SignUpScreen() {
     }
   };
 
-  const handleSubmit = () => {
+  const handleSubmit = async () => {
     if (validateName() && validateEmail()) {
+      const { error } = await supabase.auth.signInWithOtp({
```
Comment (ronniebeggs) [inline src/app/(Authentication)/SignUp/Email/index.tsx]: On the [supabase docs](https://supabase.com/docs/reference/javascript/auth-signinwithotp) I found this:

 "If the user doesn't exist, signInWithOtp() will signup the user instead. To restrict this behaviour, you can set shouldCreateUser in SignInWithPasswordlessCredentials.options to false."
 
 I imagine they have an option for verifying a OTP without officially registering a user, but I'd have to dig deeper into the docs.

File: src/app/(Authentication)/SignUp/Address/index.tsx [hunk 2]
```diff
@@ -46,7 +46,15 @@ export default function SignUpScreen() {
 
   const handleSubmit = () => {
     if (validateAddressInputs()) {
-      signUpUser(name, email, password, streetAddress, city, state, zipcode);
+      fullySignUpUser(
```
Comment (ronniebeggs) [inline src/app/(Authentication)/SignUp/Address/index.tsx:50]: Could you experiment with adding a `router.push()` call to advance the user to the AllCases screen after signup? I created an account and everything worked on the backend, but I was stuck on the address screen until I pressed back a few times. I think I just needed to clear the navigation stack to be in the correct place.

File: src/app/(Authentication)/SignUp/Email/index.tsx [hunk 3]
```diff
@@ -42,10 +43,17 @@ export default function SignUpScreen() {
     }
```
Comment (ronniebeggs) [inline src/app/(Authentication)/SignUp/Email/index.tsx:44]: Does our `validateEmail()` function check against the public users table? We could reduce a lot of OTP checks from this check alone.

File: src/app/(Authentication)/OTPFlow/index.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      <AuthInput
+        input={token}
+        onChangeInput={setToken}
+        labelText=""
+        placeholderText="Enter OTP Here"
+        isPassword={false}
+        keyboard="numeric"
+        autoCapitalization={false}
+      />
+      <TouchableOpacity
```
Comment (ronniebeggs) [inline src/app/(Authentication)/OTPFlow/index.tsx]: Add a button to resend an OTP email. I believe there's a supabase function for accomplishing this.

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Good progress so far! I say try to research further on verifying emails without immediately logging them in. I had issues where I'd verify an email, try and press back, and I could skip the password input phase. I agree that it's super annoying to test so maybe incremental progress throughout the week is best. I will look into ways to increase email input rates. Keep pushing !!!!

File: src/app/(Authentication)/SignUp/Email/index.tsx (hunk 3 above)
Comment (philipye314) [reply src/app/(Authentication)/SignUp/Email/index.tsx:44]: not right now I believe. I guess I should make a check yeah

File: src/app/(Authentication)/SignUp/Email/index.tsx (hunk 1 above)
Comment (philipye314) [reply src/app/(Authentication)/SignUp/Email/index.tsx]: added but haven't tested due to rate limiting, will work on tmrw

File: src/app/(Authentication)/SignUp/Email/index.tsx (hunk 3 above)
Comment (philipye314) [reply src/app/(Authentication)/SignUp/Email/index.tsx:44]: revisit tmrw, did a user check.

**Comment by ronniebeggs** [comment]
Final tasks:
- disable the "continue" button after password is inputted.
- display any OTP errors that may occur on screen (re-enable the continue button).
- remove the back button from the address input screen.

I will merge the auth context once there are complete.

File: src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import { useLocalSearchParams } from 'expo-router';
+import { router } from 'expo-router/src/imperative-api';
```
Comment (ronniebeggs) [inline src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx:2]: Import router from 'expo-router'

**Pull request merged**